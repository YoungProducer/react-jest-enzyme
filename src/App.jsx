import React, { useState } from 'react';

const App = ({title}) => {
    const [isTitleVisible, setIsTitleVisible] = useState(false);

    return (
        <div>
            {isTitleVisible && <h1 className="title">{title}</h1>}
            <button onClick={() => setIsTitleVisible(true)} className="show">Show title</button>
            <button onClick={() => setIsTitleVisible(false)} className="hide">Hide title</button>
        </div>
    );
};

export default App;