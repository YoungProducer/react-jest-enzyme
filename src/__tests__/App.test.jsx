import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';

import App from '../App';

describe('<App />', () => {
	const requiredProps = {
		title: 'Hello world',
	};
	// it('is matches snapshot', () => {
	//     const tree = shallow(<App />);

	//     expect(tree).toMatchSnapshot();
	// });

	// it('check props', () => {
	//     const props = {
	//         title: "Hello world"
	//     };
	//     const tree = shallow(<App {...requiredProps} />);
	//     tree.setProps(props);
	//     expect(tree.instance().test).toEqual(false);
	//     expect(tree.contains(<h1 className="title">Hello world</h1>)).toEqual(true);
	// });

	it('will title be displayed when click on "Show title"', () => {
		const onButtonClick = sinon.spy();

		const app = mount(
			<App onButtonClick={onButtonClick} {...requiredProps} />
		);

		app.find('[data-test="show-title"]').simulate('click');
		expect(app.contains(<h1 className="title">Hello world</h1>)).toEqual(
			true
		);
		app.find('[data-test="hide-title"]').simulate('click');
		expect(app.contains(<h1 className="title">Hello world</h1>)).toEqual(
			false
		);
	});
});
