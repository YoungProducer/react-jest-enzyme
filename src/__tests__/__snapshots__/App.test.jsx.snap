// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> is matches snapshot 1`] = `
<div>
  <h1>
    Hello world
  </h1>
</div>
`;
